README.md

```diff
@@ -107,6 +107,26 @@ Installing the library is as easy as:
 ember install ember-simple-auth
 ```
 
+### TypeScript setup
+1. Include `ember-cli-flash` typescript defs in the `tsconfig.json` under `paths` like this:
+   ```
+    "paths: {
+      "ember-simple-auth/*": [
+        "node_modules/ember-simple-auth"
+      ],
+    }
+   ```
+2. Import the definition for the type that you need where you need it. For eg: If you have a `session-auth.js` service that uses the `session` service offered by `ember-simple-auth`, you can import the typescript definition for it and use it like this:
+   <br /> <b>session-auth.js</b>
+   ```
+    import { inject as service } from '@ember/service';
+    import Session from 'ember-simple-auth/services/session';
+
+    export default class SessionAuthService extends Service {
+      @service declare session: Session<{}>
```

Comment:
Can we include some example SessionData here instead of the empty object?
I think the Generic here might be the most likely thing to trip up users who are adoping these type defs, so it might be useful to show an example of passing the SessionData type arg to the Session generic.

Reply:
@steveszc In our app we use the base authenticator and hence the `session.data` looks something like ```
```
{
  authenticated: {
    access_token: "super_secret_token",  // We add this
    authenticator: "authenticator:token",    // ember-simple-auth adds this
    token_type: "bearer"  //We add this
  }
}
```

This is all custom data I believe and hence I'm unsure what kind of SessionData I should use. I still a bit new to TypeScript.

Reply:
The SessionData passed to the session service in your case, if I'm not mistaken should be:

```ts
interface SessionData {
  access_token: String,
  authenticator: String,
  token_type: String,
}
```

Additionally I'm a bit torn if ESA types should allow people to just define their fields without the need for typing out `authenticator` too, since this is what ESA provides to the store.
That would be a different conversation though and wouldn't be that big of a change.
However, it is breaking down the line.
What do you think @steveszc?